src/Files.App/DataModels/NavigationControlItems/DriveItem.cs

```diff
@@ -36,15 +36,15 @@ public string Path
 			}
 		}
 
-		public string ToolTipText { get; private set; }
-		public string DeviceID { get; set; }
-		public StorageFolder Root { get; set; }
-		public NavigationControlItemType ItemType { get; set; } = NavigationControlItemType.Drive;
+		public string? ToolTipText { get; private set; }
+		public string? DeviceID { get; set; }
+		public StorageFolder? Root { get; set; }
+		public NavigationControlItemType? ItemType { get; set; } = NavigationControlItemType.Drive;
 		public Visibility ItemVisibility { get; set; } = Visibility.Visible;
 
 		public bool IsRemovable => Type == DriveType.Removable || Type == DriveType.CDRom;
 		public bool IsNetwork => Type == DriveType.Network;
-		public bool IsPinned => App.SidebarPinnedController.Model.FavoriteItems.Contains(path);
+		public bool IsPinned => App.SidebarPinnedController.Model is not null && App.SidebarPinnedController.Model.FavoriteItems.Contains(path ?? string.Empty);
```

Comment:
I'd move `path` null check before the `.Contains()` method. In case `FavoriteItems` contained an empty string, this would cause issues.
I suggest rewriting it as:

`public bool IsPinned => path is not null && App.SidebarPinnedController.Model is not null && App.SidebarPinnedController.Model.FavoriteItems.Contains(path);`